src/engine/GrbMatrix.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Herrmann (johannes.roland.herrmann@mars.uni-freiburg.de)
+
+#pragma once
+
+extern "C" {
+#include <GraphBLAS.h>
+}
+
+#include <memory>
+#include <vector>
+
+// This class wraps the functionality of the GraphBLAS object GrB_Matrix.
+// Currently only boolean matrices are supported.
+class GrbMatrix {
+ private:
+  std::unique_ptr<GrB_Matrix> matrix_;
```

Comment:
```suggestion
  std::unique_ptr<GrB_Matrix> matrix_ = std::make_unique<GrB_Matrix>();
```
Then remove the redundant code from the two constructors (the default constructor can then even be defaulted).